### ISSUE TITLE
Failed job did NOT clean the resources

### ISSUE THREAD

**Issue by luis100** [keeps/roda#968]
Crashing RODA while an ingest job was running and then restarting RODA should clean up all resources created by the job when initializing, but it failed to do so. There is an AIP (highlighted) that was created by the job that failed and still remained after RODA was initialized and the job marked as failed.

![roda - 1c02ba6b-a2ae-4b49-9108-25e5299f008e](https://cloud.githubusercontent.com/assets/631728/25616635/ebcdc916-2f35-11e7-9934-eef0ebf9580c.jpg)

![failed_job_aips_uncleaned](https://cloud.githubusercontent.com/assets/631728/25616650/f676b1c0-2f35-11e7-8a95-de5f1e3f7f79.png)

**Comment by jmaferreira** [comment]
There is a cleanup plugin. Test it please.

**Comment by luis100** [comment]
Tested it, and it failed, didn't remove the ingested AIPs, errors in log:

```
==> log/roda-core.log <==
2017-05-02 13:00:42,635 [http-apr-8080-exec-5] INFO  o.r.c.p.o.AkkaEmbeddedPluginOrchestrator - Adding job 'Remove AIP(s) whose ingest failed (1.0)' (ef16f4a8-ddcf-4919-8189-ef766766d658) to be executed
2017-05-02 13:00:42,636 [http-apr-8080-exec-5] INFO  o.r.c.p.o.AkkaEmbeddedPluginOrchestrator - Success adding job 'Remove AIP(s) whose ingest failed (1.0)' (ef16f4a8-ddcf-4919-8189-ef766766d658) to be executed
2017-05-02 13:00:42,641 [JobsSystem-st-dispatcher-26] INFO  o.r.c.p.o.akka.AkkaJobsManager - Will execute job 'ef16f4a8-ddcf-4919-8189-ef766766d658' (max: 9| exec: 1| wait: 0)
2017-05-02 13:00:42,653 [JobsSystem-io-1-dispatcher-28] DEBUG o.r.c.p.o.akka.AkkaJobStateInfoActor - Starting AkkaJobStateInfoActor router with 9 actors
2017-05-02 13:00:42,653 [JobsSystem-io-1-dispatcher-27] INFO  o.r.c.p.o.AkkaEmbeddedPluginOrchestrator - Starting Remove AIP(s) whose ingest failed (which will be done asynchronously)
2017-05-02 13:00:42,684 [JobsSystem-io-1-dispatcher-28] INFO  o.r.c.p.o.akka.AkkaJobStateInfoActor - Setting job 'Remove AIP(s) whose ingest failed (1.0)' (ef16f4a8-ddcf-4919-8189-ef766766d658) state to STARTED. Details: NO DETAILS
2017-05-02 13:00:42,721 [JobsSystem-io-1-dispatcher-28] DEBUG o.r.c.plugins.orchestrate.JobsHelper - New job completionPercentage: 100
2017-05-02 13:00:42,735 [JobsSystem-io-1-dispatcher-50] DEBUG o.r.c.plugins.orchestrate.JobsHelper - New job completionPercentage: 100
2017-05-02 13:00:42,814 [JobsSystem-io-2-dispatcher-48] ERROR o.r.c.plugins.plugins.PluginHelper - Unexpected exception during 'logic' execution
java.lang.NullPointerException: null
	at org.roda.core.index.utils.SolrUtils.appendExactMatch(SolrUtils.java:809)
	at org.roda.core.index.utils.SolrUtils.appendNotExactMatch(SolrUtils.java:994)
	at org.roda.core.index.utils.SolrUtils.parseFilterParameter(SolrUtils.java:778)
	at org.roda.core.index.utils.SolrUtils.parseFilter(SolrUtils.java:737)
	at org.roda.core.index.utils.SolrUtils.find(SolrUtils.java:278)
	at org.roda.core.index.utils.IterableIndexResult.getResults(IterableIndexResult.java:86)
	at org.roda.core.index.utils.IterableIndexResult.<init>(IterableIndexResult.java:81)
	at org.roda.core.index.IndexService.findAll(IndexService.java:181)
	at org.roda.core.index.IndexService.findAll(IndexService.java:164)
	at org.roda.core.plugins.plugins.base.CleanupFailedIngestAIPsPlugin.findAipsToDelete(CleanupFailedIngestAIPsPlugin.java:147)
	at org.roda.core.plugins.plugins.base.CleanupFailedIngestAIPsPlugin.processAIPs(CleanupFailedIngestAIPsPlugin.java:101)
	at org.roda.core.plugins.plugins.base.CleanupFailedIngestAIPsPlugin.access$000(CleanupFailedIngestAIPsPlugin.java:46)
	at org.roda.core.plugins.plugins.base.CleanupFailedIngestAIPsPlugin$1.process(CleanupFailedIngestAIPsPlugin.java:84)
	at org.roda.core.plugins.plugins.PluginHelper.processVoids(PluginHelper.java:228)
	at org.roda.core.plugins.plugins.PluginHelper.processVoids(PluginHelper.java:213)
	at org.roda.core.plugins.plugins.base.CleanupFailedIngestAIPsPlugin.execute(CleanupFailedIngestAIPsPlugin.java:79)
	at org.roda.core.plugins.orchestrate.akka.AkkaWorkerActor.handlePluginExecuteIsReady(AkkaWorkerActor.java:54)
	at org.roda.core.plugins.orchestrate.akka.AkkaWorkerActor.onReceive(AkkaWorkerActor.java:39)
	at akka.actor.UntypedActor$$anonfun$receive$1.applyOrElse(UntypedActor.scala:165)
	at akka.actor.Actor$class.aroundReceive(Actor.scala:496)
	at akka.actor.UntypedActor.aroundReceive(UntypedActor.scala:95)
	at akka.actor.ActorCell.receiveMessage(ActorCell.scala:526)
	at akka.actor.ActorCell.invoke(ActorCell.scala:495)
	at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:257)
	at akka.dispatch.Mailbox.run(Mailbox.scala:224)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
2017-05-02 13:00:42,815 [JobsSystem-io-1-dispatcher-51] DEBUG o.r.c.plugins.orchestrate.JobsHelper - New job completionPercentage: 100
2017-05-02 13:00:42,850 [JobsSystem-io-1-dispatcher-53] INFO  o.r.c.p.o.akka.AkkaJobStateInfoActor - Doing job cleanup
2017-05-02 13:00:42,861 [JobsSystem-io-1-dispatcher-53] INFO  o.r.c.p.o.akka.AkkaJobStateInfoActor - Ended doing job cleanup
2017-05-02 13:00:42,866 [JobsSystem-io-1-dispatcher-54] INFO  o.r.c.p.o.akka.AkkaJobStateInfoActor - Setting job 'Remove AIP(s) whose ingest failed (1.0)' (ef16f4a8-ddcf-4919-8189-ef766766d658) state to COMPLETED. Details: NO DETAILS
2017-05-02 13:00:42,906 [JobsSystem-st-dispatcher-26] INFO  o.r.c.p.o.akka.AkkaJobsManager - The end for job 'ef16f4a8-ddcf-4919-8189-ef766766d658' (max: 9| exec: 0| wait: 0)
```

**Comment by hsilva-keep** [comment]
Fixed since c481bb5691d8a3eb75264eefb5f7906cdd84c7f5
